kapitan/resources.py

```diff
@@ -99,6 +102,19 @@ def jinja2_render_file(search_paths, name, ctx):
 
     raise IOError("jinja2 failed to render, could not find file: {}".format(_full_path))
 
+def yaml_to_json(search_paths, name):
+    """return content of file in json"""
+    for path in search_paths:
+        _full_path = os.path.join(path, name)
+        logger.debug("yaml_to_json trying file %s", _full_path)
+        if os.path.exists(_full_path):
+            print(_full_path)
+            logger.debug("yaml_to_json found file at %s", _full_path)
+            try:
+                with open(_full_path) as f:
+                    return json.dumps(yaml.safe_load(f.read()))
+            except Exception as e:
+                raise CompileError("Parse yaml failed to parse {}: {}".format(_full_path, e))
```

Comment:
The default should be to raise IOError() like in https://github.com/deepmind/kapitan/blob/master/kapitan/resources.py#L100

Reply:
The error raised should be compile only as whether exist or not is handled in `if` condition. And in `try except` block only parsing is done.

Reply:
IOErrors could come up because of a faulty hardware as well or a bunch of other reasons other than the file missing. 
We should still check for this explicitly. It's a bit more defensive as a coding strategy but it helps debug things in the long term.